CordateLeaf/CordateLeaf.js

```diff
@@ -0,0 +1,122 @@
+import { Polar } from "../sketchlib/Polar.js";
+
+const MAX_LENGTH = 200;
+
+const RIB_ANGLE = Math.PI / 20;
+const DELAY = 8;
+const MAX_RIBS = 20;
+const UNIT_LENGTH = 1.5;
+const GROWTH_RATE = 2;
+const RIB_SCALE = 0.8;
+
+class Rib {
+  constructor(angle, growth_rate) {
+    this.angle = angle;
+    this.length = 0;
+    this.growth_rate = growth_rate;
+
+    this.children = [];
+  }
+
+  get tip() {
+    return new Polar(this.length, this.angle);
+  }
+
+  update() {
+    for (const child of this.children) {
+      child.update();
+    }
+    this.length += this.growth_rate;
+    this.length = Math.min(this.length, MAX_LENGTH);
+
+    if (this.length >= 0.5 * MAX_LENGTH && this.children.length === 0) {
+      this.children.push(new Rib(this.angle + RIB_ANGLE, this.growth_rate));
+      this.children.push(new Rib(this.angle - RIB_ANGLE, this.growth_rate));
+    }
+  }
+}
+
+function draw_rib(p, parent, rib) {
+  const tip = rib.tip;
+  p.line(
+    parent.y,
+    -parent.x,
+    UNIT_LENGTH * RIB_SCALE * tip.y,
+    UNIT_LENGTH * RIB_SCALE * -tip.x
+  );
+  p.line(
+    parent.y,
+    -parent.x,
+    UNIT_LENGTH * RIB_SCALE * -tip.y,
+    UNIT_LENGTH * RIB_SCALE * -tip.x
+  );
+
+  /*
+  for (const child of rib.children) {
+    draw_rib(p, new Polar(0.5 * tip.r, tip.theta), child);
+  }
+    */
+}
+
+export const sketch = (p) => {
+  const ribs = [];
```

Comment:
Add a comment that we only store ribs for one half of the leaf. The other half is a mirror image.